### ISSUE TITLE
Question. How good is my surrogate model?

### ISSUE THREAD

**Issue by SamiurRahman1** [interpretml/interpret-community#502]
Hi, i have seen that there is a function to calculate the r2 score of the surrogate model. I was wondering, are there any other simple metrics implemented to measure how good the surrogate model is?

Thanks

**Comment by imatiach-msft** [comment]
note that we use accuracy metric for classification scenario currently:
```
   def get_surrogate_model_replication_measure(self, training_data):
        """Return the metric which tells how well the surrogate model replicates the teacher model.
        For classification scenarios, this function will return accuracy. For regression scenarios,
        this function will return r2_score.
        :param training_data: The data for getting the replication metric.
        :type training_data: numpy.ndarray or pandas.DataFrame or scipy.sparse.csr_matrix
        :return: Metric that tells how well the surrogate model replicates the behavior of teacher model.
        :rtype: float
        """
```
but I think a lot of other metrics could be added.  I think it might even be interesting to run the surrogate model through error analysis where the "true" labels are actually the predicted labels from the teacher model to see where the surrogate model is making errors.  You can find the ErrorAnalysisDashboard here:
https://github.com/microsoft/responsible-ai-toolbox
with a lot of notebook examples here:
https://github.com/microsoft/responsible-ai-toolbox/tree/main/notebooks/individual-dashboards/erroranalysis-dashboard

**Comment by imatiach-msft** [comment]
"i have read several research papers about different metrics like stability, robustness and efficiency"
interesting, can you point to the papers specifically, maybe some of these could be implemented in this repository?  We could create issues to mark these as methods that should be implemented.

"i am trying to experiment whether the global interpretation differs when we use interpreters which are dependent on local interpreters and when we use interpreters which don't depend on local interpreter"
That sounds like really interesting research!  I'm very curious to hear what you find.

**Comment by SamiurRahman1** [comment]
here are some few example papers that talk about different evaluation methods for interpreters. the most in am interested in is number 2.

1. https://arxiv.org/abs/1906.02108
2. https://arxiv.org/abs/2008.05895
3. https://arxiv.org/abs/1910.02065

**Comment by SamiurRahman1** [comment]
i agree with your perspective. :) also, these papers are not from very good journals. but my main focus was the metrics. i am not worried about their results, rather the metrics they proposed to evaluate different interpreters. :)
